server/resp/src/main/java/org/infinispan/server/resp/Resp3Handler.java

```diff
@@ -301,8 +314,8 @@ private CompletionStage<RespRequestHandler> performMget(ChannelHandlerContext ct
                      results.set(innerCount, returnValue);
                      int length = returnValue.length;
                      if (length > 0) {
-                        // byte length + digit length (log10 + 1) + $
-                        resultBytesSize.addAndGet(returnValue.length + (int) Math.log10(length) + 1 + 1);
+                        // $ + digit length (log10 + 1) + /r/n + byte length
+                        resultBytesSize.addAndGet(1 + (int) Math.log10(length) + 1 + 2 + returnValue.length);
                      } else {
                         // $0
                         resultBytesSize.addAndGet(2);
```

Comment:
Doing a small test locally, I think handling an empty value differently is wrong, as it is missing an additional 2 bytes for the `\r\n`:

```
127.0.0.1:11222> SET key ""
OK
127.0.0.1:11222> GET key
""
127.0.0.1:11222> SET key2 value2
OK
127.0.0.1:11222> GET key2
"value2"
127.0.0.1:11222> MGET key key2
(error) ERR Server Error Encountered: writerIndex(20) + minWritableBytes(1) exceeds maxCapacity(20): PooledUnsafeDirectByteBuf(ridx: 0, widx: 20, cap: 20/20)
```

On the server:

```
2023-03-14 10:01:50,904 WARN  [o.i.s.r.RespLettuceHandler] ISPN012003: Received an unexpected exception. java.lang.IndexOutOfBoundsException: writerIndex(20) + minWritableBytes(1) exceeds maxCapacity(20): PooledUnsafeDirectByteBuf(ridx: 0, widx: 20, cap: 20/20)
	at io.netty.buffer.AbstractByteBuf.ensureWritable0(AbstractByteBuf.java:294)
	at io.netty.buffer.AbstractByteBuf.writeByte(AbstractByteBuf.java:984)
	at org.infinispan.server.resp.Resp3Handler.lambda$performMget$11(Resp3Handler.java:351)
	at org.infinispan.server.resp.RespRequestHandler.stageToReturn(RespRequestHandler.java:100)
	at org.infinispan.server.resp.RespRequestHandler.stageToReturn(RespRequestHandler.java:56)
	at org.infinispan.server.resp.Resp3Handler.performMget(Resp3Handler.java:329)
	at org.infinispan.server.resp.Resp3Handler.handleRequest(Resp3Handler.java:87)
	at org.infinispan.server.resp.RespLettuceHandler.decode(RespLettuceHandler.java:182)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:529)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:468)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:93)
	at org.infinispan.server.core.transport.StatsChannelHandler.channelRead(StatsChannelHandler.java:28)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:800)
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:499)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:397)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

If you prefer, we can handle this on another PR

Reply:
This was on main.

Reply:
This might actually be a bug related to empty string, which would be unrelated. If so, I will log a new JIRA for it.

Reply:
Okay, yes the issue is specifically with an empty string as I can't reproduce without it.

Reply:
Fixed in https://github.com/infinispan/infinispan/pull/10707